test/integrator_tests.jl

```diff
@@ -62,3 +67,9 @@ for (uprev, tprev, u, t) in intervals(DummyIntegrator())
 end
 @test eltype(collect(intervals(DummyIntegrator()))) ==
       Tuple{Vector{Float64}, Float64, Vector{Float64}, Float64}
+
+@test integrator.sol.retcode == ReturnCode.Default
+@test check_error(integrator) == ReturnCode.Success
+@test integrator.sol.retcode == ReturnCode.Default
+@test SciMLBase.check_error!(integrator) == ReturnCode.Success
+@test integrator.sol.retcode == ReturnCode.Success
```

Comment:
Add a test that nothing is allocated and this is good to go.

Reply:
Is 14f6bced0e77b3075713d56cd73d4f56448c46b4 ok? Or do you also want to test this on Lorentz using OrdinaryDiffEq?